src/SlackConnector/SlackConnection.cs

```diff
@@ -63,20 +63,26 @@ public void Initialise(ConnectionInformation connectionInformation)
             ConnectedSince = DateTime.Now;
         }
 
-        private async Task ListenTo(InboundMessage inboundMessage)
+        private Task ListenTo(InboundMessage inboundMessage)
         {
-            if (inboundMessage?.MessageType != MessageType.Message)
-                return;
-            if (string.IsNullOrEmpty(inboundMessage.User))
-                return;
-            if (!string.IsNullOrEmpty(Self.Id) && inboundMessage.User == Self.Id)
-                return;
+            if (inboundMessage == null) Task.FromResult(false);
 
-            if (inboundMessage.Channel != null && !_connectedHubs.ContainsKey(inboundMessage.Channel))
+            switch (inboundMessage.MessageType)
             {
-                _connectedHubs[inboundMessage.Channel] = _chatHubInterpreter.FromId(inboundMessage.Channel);
+                case MessageType.Message: return HandleMessage((ChatMessage)inboundMessage);
+                case MessageType.Group_Joined: return HandleGroupJoined((GroupJoinedMessage)inboundMessage);
+                case MessageType.Channel_Joined: return HandleChannelJoined((ChannelJoinedMessage)inboundMessage);
             }
 
+            return Task.FromResult(false);
```

Comment:
.NET 4.5 does not have "Task.Completed", so I have to create my own "immediately completed" task instance. It can be Task.FromResult<object>(null) or Task.FromResult(0) etc.

I would not use async (and I have removed it in a couple of places), as it makes the code more inefficient.

If a method does not have to await another async method, its recommended to return with a Task instance.

Compare these two in ildasm:
```
public Task DoSomething() {
  Console.WriteLine();
  return Task.FromResult(0);
}

public async Task DoSomething() {
  Console.WriteLine();
}
```